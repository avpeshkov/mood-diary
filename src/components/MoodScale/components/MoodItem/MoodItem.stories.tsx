import React from "react";
import { action } from "@storybook/addon-actions";
import { withKnobs, number } from "@storybook/addon-knobs";
import { MoodItem } from "./MoodItem";
export default {
    title: "MoodItem",
    decorators: [withKnobs],
};

export const nonFilled = () => [
    <MoodItem key="1" onClick={action("MoodItem clicked")} mood={number("mood", 1)} isFilled={false} />,
    <MoodItem key="2" onClick={action("MoodItem clicked")} mood={number("mood2", 7)} isFilled={false} />,
];

export const Filled = () => [
    <MoodItem key="1" onClick={action("MoodItem clicked")} mood={number("mood", 1)} isFilled={true} />,
    <MoodItem key="2" onClick={action("MoodItem clicked")} mood={number("mood2", 6)} isFilled={true} />,
];
