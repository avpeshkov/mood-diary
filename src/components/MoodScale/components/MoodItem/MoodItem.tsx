import React, { FC } from "react";
import { MoodItemButton } from "./MoodItems";

export interface MoodProps {
    onClick: () => void;
    mood: number;
    isFilled: boolean;
}

export const MoodItem: FC<MoodProps> = ({ onClick, mood, isFilled }) => {
    return <MoodItemButton mood={mood} onClick={onClick} isFilled={isFilled} />;
};
