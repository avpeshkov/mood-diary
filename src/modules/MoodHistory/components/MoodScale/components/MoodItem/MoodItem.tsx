import styled from "@emotion/styled";
import { css } from "@emotion/core";
import { Mood } from "modules/MoodHistory/types";

const BaseCell = css`
    width: 25px;
    height: 25px;
    border: 1px solid;
    display: inline-block;
    border-radius: 15px;
    line-height: 25px;
    text-align: center;
    margin: 5px;
    vertical-align: bottom;
`;

const EmptyCell = css`
    border-color: black;
`;

const filledCell = (mood: Mood) => {
    return css`
        border-color: ${MoodColorMap[mood]};
        background-color: ${MoodColorMap[mood]};
        color: gray;
    `;
};

interface Props {
    isFilled: boolean;
    mood: Mood;
}

export const MoodItem = styled.button`
    ${BaseCell};
    ${({ isFilled, mood }: Props) => (isFilled ? filledCell(mood) : EmptyCell)};
`;

export const MoodColorMap: { [key in Mood]: string } = {
    1: "#17325a",
    2: "#7223b9",
    3: "#6978dd",
    4: "#5ecacc",
    5: "#a8d765",
    6: "#f8ea55",
    7: "#f8b94f",
    8: "#ea863f",
    9: "#f86959",
    10: "#e2200c",
};
