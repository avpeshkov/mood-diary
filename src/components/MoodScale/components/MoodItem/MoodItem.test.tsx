import React from "react";
import renderer from "react-test-renderer";

import { MoodItem } from "./MoodItem";

describe("MoodItem", () => {
    it("renders not filled button", () => {
        expect(renderer.create(<MoodItem onClick={jest.fn()} mood={1} isFilled={false} />).toJSON()).toMatchSnapshot();
    });
    it("renders not filled button 2", () => {
        expect(renderer.create(<MoodItem onClick={jest.fn()} mood={5} isFilled={false} />).toJSON()).toMatchSnapshot();
    });
    it("renders filled button", () => {
        expect(renderer.create(<MoodItem onClick={jest.fn()} mood={1} isFilled={true} />).toJSON()).toMatchSnapshot();
    });
    it("renders filled button 2", () => {
        expect(renderer.create(<MoodItem onClick={jest.fn()} mood={5} isFilled={true} />).toJSON()).toMatchSnapshot();
    });
});
