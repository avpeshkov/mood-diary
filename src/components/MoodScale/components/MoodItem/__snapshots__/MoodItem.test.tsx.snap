// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoodItem renders filled button 1`] = `
.emotion-0 {
  width: 25px;
  height: 25px;
  border: 1px solid;
  display: inline-block;
  border-radius: 15px;
  line-height: 25px;
  text-align: center;
  margin: 5px;
  vertical-align: bottom;
  border-color: #372d21;
  background-color: #372d21;
  color: gray;
}

<button
  className="emotion-0"
  onClick={[MockFunction]}
/>
`;

exports[`MoodItem renders filled button 2 1`] = `
.emotion-0 {
  width: 25px;
  height: 25px;
  border: 1px solid;
  display: inline-block;
  border-radius: 15px;
  line-height: 25px;
  text-align: center;
  margin: 5px;
  vertical-align: bottom;
  border-color: #a8d765;
  background-color: #a8d765;
  color: gray;
}

<button
  className="emotion-0"
  onClick={[MockFunction]}
/>
`;

exports[`MoodItem renders not filled button 1`] = `
.emotion-0 {
  width: 25px;
  height: 25px;
  border: 1px solid;
  display: inline-block;
  border-radius: 15px;
  line-height: 25px;
  text-align: center;
  margin: 5px;
  vertical-align: bottom;
  border-color: black;
}

<button
  className="emotion-0"
  onClick={[MockFunction]}
/>
`;

exports[`MoodItem renders not filled button 2 1`] = `
.emotion-0 {
  width: 25px;
  height: 25px;
  border: 1px solid;
  display: inline-block;
  border-radius: 15px;
  line-height: 25px;
  text-align: center;
  margin: 5px;
  vertical-align: bottom;
  border-color: black;
}

<button
  className="emotion-0"
  onClick={[MockFunction]}
/>
`;
